cpp/include/cudf/reduction/detail/segmented_reduction_functions.hpp

```diff
@@ -325,6 +325,34 @@ std::unique_ptr<column> segmented_variance(column_view const& col,
                                            rmm::cuda_stream_view stream,
                                            rmm::mr::device_memory_resource* mr);
 
+/**
+ * @brief Counts the number of unique values within each segment of a column
+ *
+ * Unique entries are counted by comparing adjacent values so the column segments
+ * are expected to be sorted before calling this function otherwise the results
+ * are undefined.
+ *
+ * If input segment is empty, the segment result is null.
+ *
+ * If `null_handling==null_policy::INCLUDE`, all elements in a segment must be valid
+ * for the reduced value to be valid.
```

Comment:
is this expected behaviour for `null_handling==null_policy::INCLUDE`
Or Should it count null as unique value too?

Reply:
I believe `unique/unique_count` counts all `NULLs` as a single unique value in this case

Reply:
Updated so that `null_policy::INCLUDE` will count a null entry as a unique value in the segment.